routes/tools_routes.js

```diff
@@ -140,27 +140,16 @@ router.get('/api/topcards', async (req, res) => {
     req.logger.error(err);
     res.status(500).send({
       success: 'false',
+      numResults: 0,
+      data: [],
     });
   }
 });
 
 router.get('/topcards', async (req, res) => {
   try {
-    const { err, filter } = makeFilter(req.query.f);
-
-    if (err) {
-      req.flash('Invalid filter.');
-    }
-
-    const { data, numResults } = await topCards(filter, res);
```

Comment:
The reason it was set up this way before was to avoid having a loading spinner on initial load. I get that simplicity is good, but it doesn't seem like much code to keep that functionality in.